### ISSUE TITLE
Linux Mint 18.3 is taking the modem off-hook again!

### ISSUE THREAD

**Issue by cdgoldin** [linuxmint/linuxmint#4]
As of 15 February 2018 (and possibly earlier) the following bug has been reintroduced:

When Linux Mint (Cinnamon OR Mate) starts up (_after_ the login screen), the Smartlink internal modem is taken off hook, and remains off hook (disabling the telephone line) until such time as the system is shut down or rebooted (in which case it goes off hook again after login).

This annoying bug was introduced with Linux Mint 17.0 Trusty, and also occurs with Ubuntu 14.04. It disappeared with Linux Mint 18.0 Xenial (which is based on Ubuntu 16.04). Now it is back, possibly as a result of the latest updates, possibly earlier.

Yes, since Linux no longer supports the Smartlink modem, even with the modifications to the kernel that previously worked, I could simply disconnect the internal modem. But I dual-boot to Windoze, and DO use it without problem in Windoze, so it is more than an annoyance.

I can understand Linux not supporting hardware, but it is unacceptable that Linux is sending commands to unsupported hardware whihc cause it to malfunction.

**Comment by xenopeek** [comment]
Hi, I see you posted about your problems with Smartlink modem earlier (mid 2015) on the forums https://forums.linuxmint.com/viewtopic.php?t=200669. Linux Mint's hardware support stack comes from the used package base so yes I'd image if you have these issues on Ubuntu you'll have them on Linux Mint.

From my end I'd suggest—if you haven't already done so—to disable the ModemManager. At least I take it currently you just want to disconnect the internal modem again on Linux Mint. You can do so with the following command run from the terminal (a program you'll find in your menu):
```
sudo systemctl mask ModemManager
```
It will ask you to authorize the action with your password. Mind that on the terminal you get no visual feedback as you type a password. Just type it and press enter. The command should reply something like: "Created symlink from /etc/systemd/system/ModemManager.service to /dev/null." After running the command successfully reboot your system and see if after next login the situation is improved. If not, you can revert the above command with command `sudo systemctl unmask ModemManager` and reboot.

If disabling the ModemManager doesn't help I suggest you use the support forums to get assistance with finding which kernel module might be involved. Finding that would allow you to blacklist it and make the modem effectively inoperable. For best help include the output of the following commands in your support request on the forums, so that the knowledgeable users helping others there immediately have some information to work with:
```
inxi -Frxz
lsusb
lspci
lsmod
```
Right now your issue doesn't have enough information for a developer to work on, if indeed it would be something a developer (other than kernel or kernel module developer) could work. If none of the above suggestion solve the issue I suggest that you do take help from the support forum with troubleshooting and pinpointing the responsible component. See our [troubleshooting guide](https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/) for why I suggest that.

**Comment by cdgoldin** [comment]
Thank you for your response, but NOT for closing an unresolved issue.

Yes, I complained about this problem when it was introduced with Linux Mint 17 and Ubuntu 14.04. I never received any response, but it was finally resolved with Mint 18 and Ubuntu 16.04. Now it has resurfaced, although I'm not sure when, as I only noticed it on February 15, 2018.

Entering "sudo systemctl mask ModemManager" from a terminal (followed by the password, obviously) does absolutely nothing. It does NOT reply "Created symlink from /etc/systemd/system/ModemManager.service to /dev/null." It doesn't reply anything. And, after running the command and rebooting, nothing has changed. 

I suspect this has nothing to do with the ModemManager service, but rather the hardware detection routines, which are sending some sort of signal to the modem which it interprets as "ATDT".

>"If disabling the ModemManager doesn't help I suggest you use the support forums to get assistance with finding which kernel module might be involved...."

And which "support forums" would you suggest, since ALL of them circularly link back to (generic) GitHub with no further information?

>For best help include the output of the following commands in your support request on the forums, so that the knowledgeable users helping others there immediately have some information to work with:
inxi -Frxz
lsusb
lspci
lsmodcdg@System5 ~ $ inxi - FrxzCPU~Hexa core AMD FX-6300 Six-Core (-MCP-) speed/max~1400/3500 MHz Kernel~4.4.0-116-generic x86_64 Up~7 min Mem~598.2/7881.1MB HDD~240.1GB(32.0% used) Procs~209 Client~Shell inxi~2.2.35  cdg@System5 ~ $ lsusbBus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hubBus 007 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hubBus 006 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hubBus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hubBus 005 Device 002: ID 0764:0501 Cyber Power System, Inc. CP1500 AVR UPSBus 005 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hubBus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hubBus 004 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hubBus 011 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hubBus 010 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hubBus 009 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hubBus 008 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hubcdg@System5 ~ $ lspci00:00.0 Host bridge: Advanced Micro Devices, Inc. [AMD/ATI] RD890 PCI to PCI bridge (external gfx0 port B) (rev 02)00:02.0 PCI bridge: Advanced Micro Devices, Inc. [AMD/ATI] RD890 PCI to PCI bridge (PCI express gpp port B)00:04.0 PCI bridge: Advanced Micro Devices, Inc. [AMD/ATI] RD890 PCI to PCI bridge (PCI express gpp port D)00:05.0 PCI bridge: Advanced Micro Devices, Inc. [AMD/ATI] RD890 PCI to PCI bridge (PCI express gpp port E)00:07.0 PCI bridge: Advanced Micro Devices, Inc. [AMD/ATI] RD890 PCI to PCI bridge (PCI express gpp port G)00:11.0 SATA controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 SATA Controller [AHCI mode] (rev 40)00:12.0 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB OHCI0 Controller00:12.2 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB EHCI Controller00:13.0 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB OHCI0 Controller00:13.2 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB EHCI Controller00:14.0 SMBus: Advanced Micro Devices, Inc. [AMD/ATI] SBx00 SMBus Controller (rev 42)00:14.1 IDE interface: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 IDE Controller (rev 40)00:14.2 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] SBx00 Azalia (Intel HDA) (rev 40)00:14.3 ISA bridge: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 LPC host controller (rev 40)00:14.4 PCI bridge: Advanced Micro Devices, Inc. [AMD/ATI] SBx00 PCI to PCI Bridge (rev 40)00:14.5 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB OHCI2 Controller00:16.0 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB OHCI0 Controller00:16.2 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB EHCI Controller00:18.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h Processor Function 000:18.1 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h Processor Function 100:18.2 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h Processor Function 200:18.3 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h Processor Function 300:18.4 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h Processor Function 400:18.5 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 15h Processor Function 501:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Cedar [Radeon HD 5000/6000/7350/8350 Series]01:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Cedar HDMI Audio [Radeon HD 5400/6300 Series]02:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller (rev 09)03:00.0 USB controller: ASMedia Technology Inc. ASM1042 SuperSpeed USB Host Controller04:00.0 USB controller: ASMedia Technology Inc. ASM1042 SuperSpeed USB Host Controller05:05.0 Modem: ULi Electronics Inc. SmartLink SmartPCI563 56K Modemcdg@System5 ~ $ lsmodModule                  Size  Used bypci_stub               16384  1vboxpci                24576  0vboxnetadp             28672  0vboxnetflt             28672  0vboxdrv               462848  3 vboxnetadp,vboxnetflt,vboxpcibinfmt_misc            20480  1eeepc_wmi              16384  0asus_wmi               28672  1 eeepc_wmikvm_amd                65536  0sparse_keymap          16384  1 asus_wmivideo                  40960  1 asus_wmikvm                   548864  1 kvm_amdirqbypass              16384  1 kvmcrct10dif_pclmul       16384  0crc32_pclmul           16384  0ghash_clmulni_intel    16384  0aesni_intel           167936  0aes_x86_64             20480  1 aesni_intellrw                    16384  1 aesni_intelgf128mul               16384  1 lrwglue_helper            16384  1 aesni_intelablk_helper            16384  1 aesni_intelcryptd                 20480  3 ghash_clmulni_intel,aesni_intel,ablk_helperinput_leds             16384  0fam15h_power           16384  0edac_mce_amd           24576  0k10temp                16384  0serio_raw              16384  0edac_core              53248  0snd_hda_codec_realtek    90112  1snd_hda_codec_generic    77824  1 snd_hda_codec_realteksnd_hda_codec_hdmi     53248  1snd_hda_intel          40960  5snd_seq_midi           16384  0snd_hda_codec         135168  4 snd_hda_codec_realtek,snd_hda_codec_hdmi,snd_hda_codec_generic,snd_hda_intelsnd_seq_midi_event     16384  1 snd_seq_midisnd_hda_core           77824  5 snd_hda_codec_realtek,snd_hda_codec_hdmi,snd_hda_codec_generic,snd_hda_codec,snd_hda_intelsnd_rawmidi            32768  1 snd_seq_midisnd_hwdep              16384  1 snd_hda_codecsnd_seq                69632  2 snd_seq_midi_event,snd_seq_midisnd_pcm               106496  4 snd_hda_codec_hdmi,snd_hda_codec,snd_hda_intel,snd_hda_coresnd_seq_device         16384  3 snd_seq,snd_rawmidi,snd_seq_midisnd_timer              32768  2 snd_pcm,snd_seqsnd                    81920  21 snd_hda_codec_realtek,snd_hwdep,snd_timer,snd_hda_codec_hdmi,snd_pcm,snd_seq,snd_rawmidi,snd_hda_codec_generic,snd_hda_codec,snd_hda_intel,snd_seq_devicei2c_piix4              24576  0soundcore              16384  1 sndshpchp                 36864  0tpm_infineon           20480  08250_fintek            16384  0mac_hid                16384  0parport_pc             32768  0ppdev                  20480  0lp                     20480  0parport                49152  3 lp,ppdev,parport_pcautofs4                40960  2btrfs                 991232  0xor                    24576  1 btrfsraid6_pq              102400  1 btrfsdm_mirror              24576  0dm_region_hash         24576  1 dm_mirrordm_log                 20480  2 dm_region_hash,dm_mirrorpata_acpi              16384  0hid_generic            16384  0usbhid                 49152  1hid                   122880  2 hid_generic,usbhidamdkfd                131072  1amd_iommu_v2           20480  1 amdkfdradeon               1523712  3mxm_wmi                16384  0i2c_algo_bit           16384  1 radeonttm                    98304  1 radeonpsmouse               131072  0drm_kms_helper        155648  1 radeonsyscopyarea            16384  1 drm_kms_helpersysfillrect            16384  1 drm_kms_helpersysimgblt              16384  1 drm_kms_helperfb_sys_fops            16384  1 drm_kms_helperpata_atiixp            16384  0ahci                   36864  7drm                   364544  6 ttm,drm_kms_helper,radeonlibahci                32768  1 ahcir8169                  86016  0mii                    16384  1 r8169wmi                    20480  2 mxm_wmi,asus_wmifjes                   28672  0cdg@System5 ~ $ >Right now your issue doesn't have enough information for a developer to work on, if indeed it would be something a developer (other than kernel or kernel module developer) could work. If none of the above suggestion solve the issue I suggest that you do take help from the support forum with troubleshooting and pinpointing the responsible component. See our troubleshooting guide for why I suggest that.

It shouldn't be the user responsibility to debug the Linux operating system. I debug the programs I write. It would be nice if everyone debugged their own programs BEFORE selling them to the public.
